source-academy/frontend
# Remove editor dimensions from Redux store

### Description

See #2234 for the context behind this PR.

Fixes #2234. Part of the refactoring for #2176.

### Environment Visualizer

Note that there remains instances of `editorWidth` in `src/commons/sideContent/SideContentEnvVisualizer.tsx`:
https://github.com/source-academy/frontend/blob/40c3ea4036821a283063b5413935429fb751231a/src/commons/sideContent/SideContentEnvVisualizer.tsx#L14-L17

Searching for usages shows that whenever the `SideContentEnvVisualizer` component is used, no props are passed in:
![image](https://user-images.githubusercontent.com/5585517/195974209-70e9b35e-efcf-4018-9469-f7cd77e36876.png)

Even in #2119 where the code snippet above was introduced, the `editorWidth` prop is unused. However, I opted to leave `SideContentEnvVisualizer` unchanged because there is a mess of logic dealing with `editorWidth`:
https://github.com/source-academy/frontend/blob/40c3ea4036821a283063b5413935429fb751231a/src/commons/sideContent/SideContentEnvVisualizer.tsx#L18-L91

Testing the environment visualizer on production also shows that the current implementation doesn't really work, meaning that to remove the manual calculations would require input on the expected behaviour from the team that wrote this component:
![vmconnect_IdC4VtPBCP](https://user-images.githubusercontent.com/5585517/195974560-f2e294ff-0678-4f16-82aa-5cce1783d07a.gif)

The issue of the environment visualizer not resizing well is out of scope for this PR since none of the usages of `SideContentEnvVisualizer` even pass in the `editorWidth` prop.

### Type of change


### How to test

In any desktop workspace view (i.e., playground, mission, etc), drag the divider between the editor and the side content panel around to verify that resizing works.
